Add labels front matter documentation (#205)
grafana/writers-toolkit: docs/sources/writing-guide/front-matter/index.md

Ideally, use single terms as opposed to phrases.

### labels

Used to add labels that appear before the topic title.
Only certain labels are supported.

For `labels.products`, the supported values and the resulting website labels are:

- `cloud`: "Grafana Cloud"
- `enterprise`: "Enterprise"
- `oss`: "Open source"

Labels can be inherited through cascading front matter.

> Each project has a set of default labels defined in the root `_index.md` file for the project.

Reviewer: ```suggestion
Each project has a set of default labels that are defined in the root `_index.md` file of the project.
```

For versioned projects, the `_index.md` file resides in the website repository.

> For unversioned projects, the `_index.md` file resides in the project repository.

Reviewer: ```suggestion
For unversioned projects, the `_index.md` file resides in the project’s repository.
```

> If the default labels are incorrect for a page, or directory of pages, you should update the labels.

Reviewer 2: I feel like it would be good to say that basically, you mostly don't have to worry about labels unless you think the existing page labels are incorrect/insufficient or if you're adding a new directory.

Author: Good idea, I'll add that in.

> If the default labels are incorrect for a page, or directory of pages, you should update the labels.

Reviewer: ```suggestion
If the default labels are incorrect for a page or directory of pages, update the labels.
```

> A page should include a label in the `labels.products` sequence for every product that the page is relevant to.

Reviewer: ```suggestion
For each page, include a label in the `labels.products` sequence for every product that the page relates to.
```

> For a single page, if the page describes a feature available in Grafana Cloud and Grafana Enterprise, the source file front matter should include the following:

Reviewer 2: Are the examples in lines 96 to 113 meant to illustrate the statement made on line 94?

Author: Yeah, that was my intent :)

Reviewer 2: Okay, then maybe just add a bit of connective tissue like so.
```suggestion
For example, if a **single page** describes a feature available in Grafana Cloud and Grafana Enterprise, the source file front matter should include the following:
```

> For a single page, if the page describes a feature available in Grafana Cloud and Grafana Enterprise, the source file front matter should include the following:

Reviewer: ```suggestion
For a single page, if the page describes a feature available in Grafana Cloud and Grafana Enterprise, the source file front matter looks as follows:
```

Author: My concern with the suggested wording is that the following snippet is only part of the front matter. I would expect all of the other usual front matter mappings to exist in addition to the snippet example.

```yaml
labels:
  products:
    - cloud
    - enterprise
```

> For a directory of pages, if all pages describe a feature only available in Grafana Cloud, the branch bundle `_index.md` file front mattershould include the following:

Reviewer 2: ```suggestion
For a directory of pages, if all pages describe a feature only available in Grafana Cloud, the branch bundle `_index.md` file front matter should include the following:
```

> For a directory of pages, if all pages describe a feature only available in Grafana Cloud, the branch bundle `_index.md` file front mattershould include the following:

Reviewer: ```suggestion
For a directory of pages, if all pages describe a feature that is only available in Grafana Cloud, the branch bundle `_index.md` file front matter looks as follows:
```

> ```

Reviewer 2: Should line 107 also say "yaml" like in the example above?
